File: 1910.remove-all-occurrences-of-a-substring.py
#
# @lc app=leetcode id=1910 lang=python3
#
# [1910] Remove All Occurrences of a Substring
#

# @lc code=start
class Solution:
    '''
    1 <= s.length <= 1000
    1 <= part.length <= 1000
    So just use stack
    '''
    def removeOccurrences(self, s: str, part: str) -> str:
        result = []
        n = len(part)
        for c in s:
            result.append(c)
            if len(result) >= n and ''.join(result[-n::]) == part:
                for j in range(n):
                    result.pop()
        return ''.join(result)

# @lc code=end

File: 1910.remove-all-occurrences-of-a-substring.cpp
/*
 * @lc app=leetcode id=1910 lang=cpp
 *
 * [1910] Remove All Occurrences of a Substring
 */

// @lc code=start
#include <string>
using namespace std;

class Solution {
public:
    string removeOccurrences(string s, string part) {
        string result;
        int n = part.size();
        
        for (char c : s) {
            result.push_back(c);
            
            if (result.size() >= n && result.substr(result.size() - n) == part) {
                for (int j = 0; j < n; j++) {
                    result.pop_back();
                }
            }
        }
        
        return result;
    }
};

// @lc code=end

